### PR TITLE
Incorporates the creation of Delaunay Lines in the API

### PR DESCRIPTION
**Pull request by pedrocamargo**
Facilitates the creation of multiple Delaunay Lines, as well as simplify the generation of these in QGIS

### CODE REVIEW

**Comment by jamiecook** [comment]
did you mean to include sioux falls zip file?

**Comment by pedrocamargo** [comment]
Yes, I did.  It has been there for a while as one of the examples people can create. It happened to have data that needed to be removed, though

File: aequilibrae/utils/create_delaunay_network.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+        edges = pd.DataFrame(edges)
+        edges.drop_duplicates(inplace=True)
+        edges.columns = ['a_node', 'b_node']
+        edges = edges.assign(direction=0, distance=0, link_id=np.arange(edges.shape[0]) + 1)
+        edges = edges[['link_id', 'direction', 'a_node', 'b_node', 'distance']]
+        edges.to_sql(DELAUNAY_TABLE, conn, index=False)
+
+        # Now we create the geometries for the delaunay triangulation
+        conn.execute("select AddGeometryColumn( 'delaunay_network', 'geometry', 4326, 'LINESTRING', 'XY', 0);")
```
Comment (jamiecook) [inline aequilibrae/utils/create_delaunay_network.py:65]: hardcoding of your constant (DELAUNAY_TABLE)

File: aequilibrae/utils/create_delaunay_network.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        ta.execute()
+
+        report = {"setup": str(ta.info())}
+        data = [result_name, "Delaunay assignment", self.procedure_id, str(report), ta.procedure_date, '']
+        conn.execute("""Insert into results(table_name, procedure, procedure_id, procedure_report, timestamp,
+                                            description) Values(?,?,?,?,?,?)""", data)
+        conn.commit()
+        conn.close()
+
+        cols = []
```
Comment (jamiecook) [inline aequilibrae/utils/create_delaunay_network.py:118]:
```suggestion
        cols = [x for v in matrix.view_names for x in [f'{v}_ab', f'{v}_ba', f'{v}_tot']]
```

I do love me some list comps

File: tests/aequilibrae/utils/test_create_delaunay_network.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    def setUp(self) -> None:
+        self.proj = create_example(join(gettempdir(), uuid4().hex))
+
+    def tearDown(self) -> None:
+        self.proj.close()
+
+    def test_create_delaunay_network(self):
+        da = DelaunayAnalysis()
+        with self.assertRaises(ValueError):
+            da.create_network('nodes')
```
Comment (jamiecook) [inline tests/aequilibrae/utils/test_create_delaunay_network.py:19]: did you know you can do this?

```suggestion
        self.assertRaises(ValueError, da.create_network, 'nodes'):
```

File: tests/aequilibrae/utils/test_create_delaunay_network.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+        da.create_network()
+        self.assertEqual(59, self.proj.conn.execute('select count(*) from delaunay_network').fetchone()[0])
+
+        da.create_network('network', True)
+        self.assertEqual(62, self.proj.conn.execute('select count(*) from delaunay_network').fetchone()[0])
+
+        with self.assertRaises(ValueError):
```
Comment (jamiecook) [inline tests/aequilibrae/utils/test_create_delaunay_network.py:27]:
```suggestion
       self.assertRaises(ValueError, da.create_network):
```

**Review by jamiecook — APPROVED** [review]

**Pull request merged**